tests/utils/test_url.py

```diff
@@ -0,0 +1,90 @@
+import unittest
+from unittest.mock import MagicMock, patch
+
+from page_analyzer.utils.url import (
+    check_url,
+    normalize,
+    parse_page_data,
+    validate,
+)
+
+
+def test_validate():
+    # given
+    url = "https://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    assert errors == {}
+
+
+def test_validate_invalid():
+    # given
+    url = "htttps://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    expected_errors = {"name": "Некорректный URL"}
+    assert errors == expected_errors
+
+
+def test_normalize():
+    # given
+    url = "http://www.example.com:1030/software/index.html"
+
+    # when
+    normalized_url = normalize(url)
+
+    # then
+    expected_url = "http://www.example.com"
+    assert normalized_url == expected_url
+
+
+class TestCheckUrl(unittest.TestCase):
+    @patch("page_analyzer.utils.url.parse_page_data")
+    @patch("page_analyzer.utils.url.requests")
+    def test_check_url(self, mock_requests, mock_parse_page_data):
+        request_response_mock = MagicMock()
+        request_response_mock.status_code = 200
+        parsing_data = {
+            "response_code": "200",
+            "h1": "some header",
+            "title": "some title",
+            "description": "some description",
+        }
+        mock_requests.get.return_value = request_response_mock
+        mock_parse_page_data.return_value = parsing_data
+        self.assertEqual("some header", check_url("https://example.com")["h1"])
+
+    @patch("page_analyzer.utils.url.requests")
+    def test_check_url_exception(self, mock_requests):
+        mock_requests.get.side_effect = Exception
+        with self.assertRaises(Exception):
+            check_url("http://www.fake.com")
+
+    @patch("page_analyzer.utils.url.requests")
+    def test_check_url_unavailable(self, mock_requests):
+        request_response_mock = MagicMock()
+        request_response_mock.status_code = 404
+        mock_requests.get.return_value = request_response_mock
+        self.assertIsNone(check_url("http://www.unavailable.com"))
+
+
+class TestParsePageData(unittest.TestCase):
+    @patch("page_analyzer.utils.url.requests")
+    def test_parse_page_data(self, mock_requests):
+        request_response_mock = MagicMock()
+        mock_requests.get.return_value = request_response_mock
+        html_content = (
+            "<title>title</title>"
+            "<meta name='description' content='some description'></meta>"
+            "<h1>header</h1>"
+        )
+        request_response_mock.content = html_content
+        self.assertEqual(
+            "some description", parse_page_data(request_response_mock)["description"]
+        )
```

Comment:
wdyt about checking the whole returned object, not only one key? describing the full object will better describe code behaviour